configure/os/CONFIG.Common.RTEMS

```diff
@@ -107,6 +107,7 @@ OP_SYS_CFLAGS += $(OP_SYS_CFLAGS_NET_$(RTEMS_HAS_NETWORKING))
 
 ifeq ($(RTEMS_HAS_POSIX_API),yes)
 POSIX_CPPFLAGS = -D_GNU_SOURCE -D_DEFAULT_SOURCE
+POSIX_CPPFLAGS += -DFD_SETSIZE=150
```

Comment:
I'm not sure what we did on RTEMS-4, but 150 is not a multiple of 4. Doesn't `select()` map FDs to individual bits in the memory buffer? If so wouldn't it be better to make this a multiple of 8 or 32?

Reply:
No need to use the past tense here...

https://github.com/epics-base/epics-base/blob/fe9995c0b50e0aacd43e7aaa7fe6fab1e866db2a/modules/libcom/RTEMS/score/rtems_config.c#L36

I agree that 150 seems arbitrary.  As were the previous values of 50 and 100.  08297f56b385a7ff1517dc854c67fb536b508418  46068e813dc7333810411f1b156bc89e4ffb83d7  deae06a694dd31c06cacca4e04b9f41100c4e548

More significant than bits in `fd_set`, `CONFIGURE_MAXIMUM_FILE_DESCRIPTORS` is sizing the `rtems_libio_iops[]` array.  I suppose we could try to tie this setting to `sizeof(rtems_libio_t)`, but I suspect it would not be worth the effort.

Reply:
I have no used this option. If it works as documented it is good solution for applications that define the number of descriptors and can control the `FD_SETSIZE` for all `select` calls. For libraries built and used by EPICS that do not have access to the number of descriptors this does not work and the [now standard work](https://git.rtems.org/rtems/tree/cpukit/mghttpd/mongoose.c#n1525) around in the code is needed.

Note, in libbsd the [`select` kernel call](https://git.rtems.org/rtems-libbsd/tree/freebsd/sys/kern/sys_generic.c?h=6-freebsd-12#n998) locally allocates 2048 bits which should be more than enough.

Reply:
Core Group:
@kiwichris This approach should work with newlib (implementation is messy).
ANJ: Please make the 150 a build variable that can be overridden in the BSP's CONFIG_SITE file.